modules/articles.py

```diff
@@ -0,0 +1,59 @@
+import newspaper
+import re
+
+class article_content():
+
+    def __init__(self, article_object):
+        #Whenever a class object is made, the article is tokenized and the dictionary file is opened as a set.
+        self.tokenized_words = nltk.word_tokenize(article_object.text)
+        self.article_object = article_object
+        with open('words_alpha.txt') as word_file:
+            self.valid_words = set(word_file.read().split())
+        
+        self.misspells = 0  # Set initial misspells to 0.
+
+    def citation_check(self):
+        
+        #Something NLTK stop words. These often aren't real citations. / to remove local files, and ad removal from scraped link.
+        self.not_citations = ("facebook.com", "twitter.com", "#", "mailto", "plus.google", "advertising", "advertisement", "ad.", "/", "whatsapp", "quora", "nav.", "instagram")
+
+        self.article_html = self.article_object.html
+        http = [self.article_object.url] + re.findall(r'href="[^\"]+', self.article_html) #Regex code for finding links, eg href="http://www.bbc.com" returns http://www.bbc.com.
```

Comment:
Soup would be faster at this.

But Im quite sure you could get citations and urls from the newspaper module.

Reply:
Trying to keep imported libraries to minimum.

Soup would easily give the citations with 
```
for link in soup.find_all('a'):
    print(link.get('href'))
```
But it would pull stuff like "https://github.com/Learning-Python-Team/UnTruth/pull/20" but I consider each website source linked as only one citation, so want "github", so regex would be needed anyway to detect the / (and remove everything ahead) and only retain "github.com".

I would have to use soup.findAll(re.compile("REGEX EXPRESSION")) at which point, I think directly using regex is probably better. I'll compare the timings and see later on when I have time. 

> But Im quite sure you could get citations and urls from the newspaper module.

I could not find this in newspaper quickstart documentation.

Reply:
Yeah, I did some googling, but I think I was wrong regarding newspaper returning links. However, newspaper uses the requests library. With this library, you can easily get all links quicker than bs4.
A quick example would be;
`session.get(URL).html.absolute_links` With the list that this returns, you could use the urlparse library (or plain regex) to categorize the links. I'd say, give it a quick test to see which is faster and go for the fastest option.

Reply:
@mandjevant how do one check if one is faster over the other ?

Reply:
On a CPU, even for a preloaded requests_html session and parsed article, the regex method is faster.

![image](https://user-images.githubusercontent.com/22197137/64477907-78ab6280-d1be-11e9-96d8-5ac458096bb8.png) by a order of 1000 or so (microseconds vs milliseconds).

But we'll need the parsed article anyway for other functions and the whole article analysis so if you consider loading the requests_html within the cell, it's even worse.

![image](https://user-images.githubusercontent.com/22197137/64477938-ef486000-d1be-11e9-9d4d-090503dde96b.png)


Timed over a 100 loops - Still quite better (even with the session made outside the block)
![image](https://user-images.githubusercontent.com/22197137/64478006-ab098f80-d1bf-11e9-96ee-daad19ac7f43.png)

Colab notebook on which I did the testing - https://colab.research.google.com/drive/1o94PRcDUXiARfpt3fYeMiCw_OVm3J94h

Reply:
As for further improvement, I'm going to count citations that are only within the <body> tags, because outside that it's probably ads and navigation and footer. What do you think @mandjevant ? Is that wise to do?

Reply:
> 
> 
> how do one check if one is faster over the other ?

You can use time or timeit. https://docs.python.org/2/library/timeit.html
If you are familiar with Jupyter notebooks, they have a shortcut for it. 

%time for one line time. 
%timeit for one line time, iterated over many loops (like I did for 100 in above image) and finding the best 3 of them  (usually)

%%time for the entire cell time.
%%timeit for the entire cell time, iterated over many loops, and finding the best 3 of them (usually).